frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.tsx

```diff
@@ -139,85 +133,51 @@ function ActionCreatorComponent({
     );
   };
 
+  const saveModalElement = showSaveModal ? (
+    <Modal title={t`New Action`} onClose={handleCloseNewActionModal}>
+      <CreateActionForm
+        question={question}
+        formSettings={formSettings as ActionFormSettings}
+        modelId={defaultModelId}
+        onCreate={handleSave}
+        onCancel={handleCloseNewActionModal}
+      />
+    </Modal>
+  ) : null;
+
   return (
-    <>
-      <Modal wide onClose={onClose}>
-        <ModalRoot>
-          <ActionCreatorBodyContainer>
-            <ModalLeft>
-              <DataReferenceTriggerButton onClick={toggleDataRef} />
-              <ActionCreatorHeader
-                type="query"
-                name={question.displayName() ?? t`New Action`}
-                onChangeName={newName =>
-                  setQuestion(q => q.setDisplayName(newName))
-                }
-              />
-              <EditorContainer>
-                <QueryActionEditor
-                  question={question}
-                  setQuestion={setQuestion}
-                />
-              </EditorContainer>
-              <ModalActions>
-                <Button onClick={onClose} borderless>
-                  {t`Cancel`}
-                </Button>
-                <Button
-                  primary
-                  disabled={query.isEmpty()}
-                  onClick={handleSaveClick}
-                >
-                  {isNew ? t`Save` : t`Update`}
-                </Button>
-              </ModalActions>
-            </ModalLeft>
-
-            <DataReferenceInline
-              isOpen={isDataRefOpen}
-              onClose={closeDataRef}
-            />
-
-            {!isDataRefOpen && (
-              <FormCreator
-                params={question?.parameters() ?? []}
-                formSettings={
-                  question?.card()?.visualization_settings as ActionFormSettings
-                }
-                onChange={setFormSettings}
-                onExampleClick={handleExampleClick}
-              />
-            )}
-          </ActionCreatorBodyContainer>
-        </ModalRoot>
-      </Modal>
-      {showSaveModal && (
-        <Modal title={t`New Action`} onClose={handleClose}>
-          <CreateActionForm
-            question={question}
-            formSettings={formSettings as ActionFormSettings}
-            modelId={defaultModelId}
-            onCreate={handleOnSave}
-            onCancel={handleClose}
-          />
-        </Modal>
-      )}
-    </>
+    <ActionCreatorView
+      isNew={isNew}
+      isDataRefOpen={isDataRefOpen}
+      disableSaveButton={query.isEmpty()}
+      question={question}
+      saveModalElement={saveModalElement}
+      onToggleDataRef={toggleDataRef}
+      onCloseDataRef={closeDataRef}
+      onChangeQuestionQuery={handleChangeQuestionQuery}
+      onChangeName={newName =>
+        setQuestion(question => question.setDisplayName(newName))
+      }
+      onCloseModal={onClose}
+      onChange={setFormSettings}
```

Comment:
Should we specify which changes do we expect here? Maybe call it `onFormSettingsChange`?

Reply:
I'd name it `onChangeFormSettings` to align with our naming convention at least within this file. Following
> The name change follows other callback names in this file which is in this format `handle + action[ + optional modifier]` e.g. `onChangeName`

[Ref](https://github.com/metabase/metabase/pull/27832#discussion_r1084430644)

Reply:
Done!